iothub/device/src/MethodRequest.cs

```diff
@@ -6,20 +6,46 @@
 
 namespace Microsoft.Azure.Devices.Client
 {
+    /// <summary>
+    /// The data structure that represents a method request.
+    /// </summary>
     public sealed class MethodRequest
     {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="MethodRequest"/> class with without any method data
+        /// and extra time for device to connect and send a response. The default timeout is 15 seconds.
+        /// </summary>
+        /// <param name="name">The method name.</param>
         public MethodRequest(string name) : this(name, null, null, null)
         {
         }
 
+        /// <summary>
+        /// Initializes a new instance of the <see cref="MethodRequest"/> class.
+        /// </summary>
+        /// <param name="name">The method name.</param>
+        /// <param name="data">The method data.</param>
         public MethodRequest(string name, byte[] data) : this(name, data, null, null)
         {
         }
 
+        /// <summary>
+        /// Initializes a new instance of the <see cref="MethodRequest"/> class without any method data.
+        /// </summary>
+        /// <param name="name">The method name.</param>
+        /// <param name="responseTimeout">The method timeout value.</param>
+        /// <param name="connectionTimeout">The device connection timeout value.</param>
         public MethodRequest(string name, TimeSpan? responseTimeout, TimeSpan? connectionTimeout) : this(name, null, responseTimeout, connectionTimeout)
         {
         }
 
+        /// <summary>
+        /// Initializes a new instance of the <see cref="MethodRequest"/> class.
+        /// </summary>
+        /// <param name="name">The method name.</param>
+        /// <param name="data">The method data.</param>
+        /// <param name="responseTimeout">The method timeout value.</param>
+        /// <param name="connectionTimeout">The device connection timeout value.</param>
```

Comment:
It's worth spelling out what these response/connect timeouts actually do. You can read about it in the paragraph in [this doc](https://docs.microsoft.com/en-us/azure/iot-hub/iot-hub-devguide-direct-methods#method-invocation)

Reply:
Also worth adding that there are service side limits to these values, also found in the above doc